packages/smooth_app/lib/pages/product/nutrition_page_loaded.dart

```diff
@@ -0,0 +1,281 @@
+        }
+      }
+      children.add(
+        ElevatedButton.icon(
+          onPressed: () async {
+            final List<OrderedNutrient> availables = <OrderedNutrient>[];
+            for (final OrderedNutrient orderedNutrient in _displayableList) {
+              final String id = orderedNutrient.id;
+              final dynamic value = _getValue(id);
+              final bool addAble = value == null && !orderedNutrient.important;
+              if (addAble) {
+                availables.add(orderedNutrient);
+              }
+            }
+            availables.sort(
+                (final OrderedNutrient a, final OrderedNutrient b) =>
+                    a.name!.compareTo(b.name!));
+            final OrderedNutrient? selected = await showDialog<OrderedNutrient>(
+                context: context,
+                builder: (BuildContext context) {
+                  final List<Widget> children = <Widget>[];
+                  for (final OrderedNutrient nutrient in availables) {
+                    children.add(
+                      ListTile(
+                        title: Text(nutrient.name!),
+                        onTap: () => Navigator.pop(context, nutrient),
+                      ),
+                    );
+                  }
+                  return AlertDialog(
+                    title: Text(appLocalizations.nutrition_page_add_nutrient),
+                    content: SizedBox(
+                      // TODO(monsieurtanuki): proper sizes
+                      width: 300,
+                      height: 400,
+                      child: ListView(children: children),
+                    ),
+                    actions: <Widget>[
+                      ElevatedButton(
+                        onPressed: () => Navigator.pop(context),
+                        child: Text(appLocalizations.cancel),
+                      ),
+                    ],
+                  );
+                });
+            if (selected != null) {
+              setState(() => _initValue(selected.id));
+            }
+          },
+          icon: const Icon(Icons.add),
+          label: Text(appLocalizations.nutrition_page_add_nutrient),
+        ),
+      );
+    }
+
+    return Scaffold(
+      appBar: AppBar(title: Text(appLocalizations.nutrition_page_title)),
+      body: Padding(
+        padding: const EdgeInsets.all(8.0),
+        child: ListView(children: children),
+      ),
+    );
+  }
+
+  void _populateOrderedNutrientList(final List<OrderedNutrient>? list) {
+    if (list == null) {
+      return;
+    }
+    for (final OrderedNutrient nutrient in list) {
+      _displayableList.add(nutrient);
+      _populateOrderedNutrientList(nutrient.subNutrients);
+    }
+  }
+
+  Widget? _getNutrientWidget(final OrderedNutrient orderedNutrient) {
+    final String id = orderedNutrient.id;
+    final dynamic value = _getValue(id);
+    if (value == null && !orderedNutrient.important) {
+      return null;
+    }
```

Comment:
We don't allow editing of nutrients when they are not important?

Reply:
There are nutrients displayed by default, and other nutrients that you can add on an ad-hoc basis.
Of course, other nutrients should be displayed if we already have a value for them.

Reply:
Yes we do: if the nutrient is not in the displayed list, you can add it. But by default we display only the important and populated nutrients.

Reply:
Sounds good